pkg/kubelet/cm/device_plugin_handler.go

```diff
@@ -136,27 +151,57 @@ func NewDevicePluginHandlerImpl(updateCapacityFunc func(v1.ResourceList)) (*Devi
 		updateCapacityFunc(capacity)
 	}
 
-	mgr, err := deviceplugin.NewManagerImpl(pluginapi.KubeletSocket, deviceManagerMonitorCallback)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to initialize device plugin manager: %+v", err)
-	}
-
-	handler.devicePluginManager = mgr
-	handler.devicePluginManagerMonitorCallback = deviceManagerMonitorCallback
+	handler.devicePluginMonitorCallback = devicePluginMonitorCallback
 	// Loads in allocatedDevices information from disk.
-	err = handler.readCheckpoint()
+	err := handler.readCheckpoint()
 	if err != nil {
 		glog.Warningf("Continue after failing to read checkpoint file. Device allocation info may NOT be up-to-date. Err: %v", err)
 	}
 	return handler, nil
 }
 
 func (h *DevicePluginHandlerImpl) Start() error {
-	return h.devicePluginManager.Start()
+	glog.V(2).Infof("Starting Device Plugin Handler")
+
+	socketPath := filepath.Join(h.socketdir, h.socketname)
+	os.MkdirAll(h.socketdir, 0755)
+
+	// Removes all stale sockets in h.socketdir. Device plugins can monitor
+	// this and use it as a signal to re-register with the new Kubelet.
+	if err := h.removeContents(h.socketdir); err != nil {
+		glog.Errorf("Fail to clean up stale contents under %s: %+v", h.socketdir, err)
+	}
+
+	if err := os.Remove(socketPath); err != nil && !os.IsNotExist(err) {
```

Comment:
Is this necessary ? Didn't we just removed all sockets with `removeContents` ?

Reply:
good catch